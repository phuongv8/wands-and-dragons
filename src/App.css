.App {
  text-align: center;
}

#root {
  height: 100%;
}

a {
  text-decoration: none;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.search-result {
  position: absolute;
  display: none;
  left: 0;
  right: 0%;
  top: 65px;
  z-index: 15;
  color: #fff;
  font-size: 20px;
  overflow: scroll;
  padding: 15px;
  justify-content: center;
  transition: all 0.5s ease-in-out;
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;
  -webkit-transition: all 0.5s ease-in-out;
  animation-name: collapse;
  -webkit-animation-name: collapse;
  -webkit-backdrop-filter: blur(8px);
  backdrop-filter: blur(8px);
  box-shadow: 0px 10px 15px 10px rgba(0, 0, 0, 0.15);
  background-color: rgba(0, 0, 0, 0.849);
}

.search-result.active {
  display: flex;
  max-height: 500px;
  transition: all 0.5s ease-in-out;
  -webkit-transition: all 0.5s ease-in-out;
  animation-name: expand;
  -webkit-animation-name: expand;
  padding: 25px;
}

@keyframes expand {
  0% {
    max-height: 0;
  }
  100% {
    max-height: calc(50% - 48px);
  }
}

@keyframes collapse {
  0% {
    height: calc(100% - 48);
  }
  100% {
    height: 0;
    display: none;
  }
}

.search-result.active .inner {
  height: 100%;
}

.product-line-divider {
  width: 100%;
  display: block;
  border-bottom: solid 2px #6f6c6c;
  margin: 10px 0;
}

.navbar-logo {
  height: 50px;
}

.navbar-logo img {
  height: 100%;
}

.header-main {
  height: 75px;
  display: flex;
}

.top-nav-section-container {
  max-width: 1440px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
}

.bottom-nav-section-container {
  width: 100%;
  border-bottom: var(--bs-border-width) var(--bs-border-style)
    var(--bs-border-color) !important;
}

.bottom-nav-section-container {
  width: 100%;
  background-color: #f8f9fa;
}

.error-msg {
  color: red;
}

.bottom-nav-section-container .navbar {
  max-width: 1440px;
  margin: 0 auto;
  min-width: 380px;
}

.navbar > .container,
.navbar > .container-fluid,
.navbar > .container-lg,
.navbar > .container-md,
.navbar > .container-sm,
.navbar > .container-xl,
.navbar > .container-xxl {
  display: flex;
  flex-wrap: inherit;
  align-items: center;
  justify-content: space-between;
  justify-content: end;
}

.badge > a:nth-child(1) > svg:nth-child(1) > path:nth-child(1) {
  color: black;
}

.container,
.container-fluid,
.container-lg,
.container-md,
.container-sm,
.container-xl,
.container-xxl {
  width: inherit;
}

.container,
.container-fluid,
.container-lg,
.container-md,
.container-sm,
.container-xl,
.container-xxl {
  margin-left: inherit;
}

div.container-fluid {
  margin-left: auto;
}

.navbar > .container,
.navbar > .container-fluid,
.navbar > .container-lg,
.navbar > .container-md,
.navbar > .container-sm,
.navbar > .container-xl,
.navbar > .container-xxl {
  margin-right: 0;
}

.css-1c32n2y-MuiBadge-root {
  margin-right: 20px;
}

.navbar > .container,
.navbar > .container-fluid,
.navbar > .container-lg,
.navbar > .container-md,
.navbar > .container-sm,
.navbar > .container-xl,
.navbar > .container-xxl {
  margin: 0;
}

#navbarNavDropdown li {
  color: #fff;
  letter-spacing: 0.26em;
  padding-bottom: 15px;
}

.navbar-nav {
  margin: 50px 0;
}

#navbarNavDropdown {
  position: absolute;
  top: 69px;
  z-index: 25;
  left: 0px;
  right: 0px;
  justify-content: center;
  text-align: center;
  -webkit-backdrop-filter: blur(8px);
  backdrop-filter: blur(8px);
  box-shadow: 0px 10px 15px 10px rgba(0, 0, 0, 0.15);
  background-color: rgba(0, 0, 0, 0.849);
}

.form-inputs > div:nth-child(1) > input:nth-child(1) {
  width: 230px;
}

@media (min-width: 768px) {
  div.container-fluid:nth-child(2) {
    margin-left: auto;
  }
}

@media (min-width: 992px) {
  #navbarNavDropdown {
    display: flex;
    position: relative;
    justify-content: center;
    text-align: center;
    background-color: inherit;
    box-shadow: unset;
    backdrop-filter: unset;
  }

  #navbarNavDropdown li {
    color: inherit;
  }

  .navbar-nav {
    margin: 0;
  }

  #navbarNavDropdown {
    top: 0;
  }

  div.container-fluid:nth-child(1) {
    justify-self: first baseline;
    margin-right: auto;
  }

  #navbarNavDropdown li {
    color: inherit;
    letter-spacing: 0.25em;
    margin: 0 15px;
  }
}
