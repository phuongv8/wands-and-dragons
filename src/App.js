import React from "react";
import { Routes, Route } from "react-router-dom";
import "./App.css";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";
import HomePage from "./views/HomePage";
import ProductsPage from "./views/ProductsPage";
import CartPage from "./views/CartPage";
import ProductDetailPage from "./views/ProductDetailPage";
import AdminPage from "./views/AdminPage";
import OrderHistoryPage from "./views/OrderHistoryPage";
import { AuthProvider } from "./components/Auth";
import { Login } from "./views/Login";
import { RequireAuth } from "./components/RequireAuth";
import { AboutPage } from "./views/AboutPage";
import { ProductsProvider } from "./utils/ProductProvider";
import { ConfirmationPage } from "./views/ConfirmationPage.jsx";
function App() {
  return (
    <AuthProvider>
      <ProductsProvider>
        <React.Fragment>
          <Navbar />
          <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path="/productspage" element={<ProductsPage />} />
            <Route path="category/:type" element={<ProductsPage />} />
            <Route path="productdetail" element={<ProductDetailPage />} />
            <Route path="cart" element={<CartPage />} />
            <Route path="loginauth" element={<Login />} />
            <Route path="about" element={<AboutPage />} />
            <Route path="orders" element={<OrderHistoryPage />} />
            <Route path="confirmation" element={<ConfirmationPage />} />
            <Route
              path="inventory"
              element={
                <RequireAuth>
                  <AdminPage />
                </RequireAuth>
              }
            />
          </Routes>
          <Footer />
        </React.Fragment>
      </ProductsProvider>
    </AuthProvider>
  );
}

export default App;
