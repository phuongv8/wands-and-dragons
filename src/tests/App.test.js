import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import App from "../App.js";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders app.js correctly", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const view = renderer.getRenderOutput();

  expect(view).toMatchSnapshot();
});
