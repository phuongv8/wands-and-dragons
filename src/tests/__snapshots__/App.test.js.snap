// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app.js correctly 1`] = `
<AuthProvider>
  <ProductsProvider>
    <React.Fragment>
      <Navbar />
      <Routes>
        <Route
          element={<HomePage />}
          path="/"
        />
        <Route
          element={<ProductsPage />}
          path="/productspage"
        />
        <Route
          element={<ProductsPage />}
          path="category/:type"
        />
        <Route
          element={<ProductDetailPage />}
          path="productdetail"
        />
        <Route
          element={<CartPage />}
          path="cart"
        />
        <Route
          element={<Login />}
          path="loginauth"
        />
        <Route
          element={<OrderHistoryPage />}
          path="orders"
        />
        <Route
          element={
            <RequireAuth>
              <AdminPage />
            </RequireAuth>
          }
          path="inventory"
        />
      </Routes>
    </React.Fragment>
  </ProductsProvider>
</AuthProvider>
`;
